code/data_structures/src/stack/infix_to_postfix/infix_to_postfix_1.cpp

```diff
@@ -0,0 +1,124 @@
+            stack1.push(expression_[i]);
+
+            //if scanned character is opened bracket pop all literals from stack till matching open bracket gets poped
+        else if(expression_[i] == ')' || expression_[i] == ']' || expression_[i] == '}')
+        {
+            if(expression_[i] == ')')
+            {
+                while(stack1.top() != '(')
+                {
+                    infixToPostfixExp = infixToPostfixExp + stack1.top();
+                    stack1.pop();
+                }
+            }
+            if(expression_[i] == ']')
+            {
+                while(stack1.top() != '[')
+                {
+                    infixToPostfixExp = infixToPostfixExp + stack1.top();
+                    stack1.pop();
+                }
+            }
+            if(expression_[i] == '}')
+            {
+                while(stack1.top() != '{')
+                {
+                    infixToPostfixExp = infixToPostfixExp + stack1.top();
+                    stack1.pop();
+                }
+            }
+            stack1.pop();
+        }
+            //if scanned character is operator
+        else if(expression_[i] == '+' || expression_[i] == '-' || expression_[i] == '*' || expression_[i] == '/' || expression_[i] == '^')
+        {
+            //very first operator of expression is to be pushed on stack
+            if(stack1.empty()) {
+                stack1.push(expression_[i]);
+            }
+            else{
+                /*
+                 * check the precedence order of instack(means the one on top of stack) and incoming operator,
+                 * if instack operator has higher priority than incoming operator pop it out of stack&put it in
+                 * final postifix expression, on other side if precedence order of instack operator is less than i
+                 * coming operator, push incoming operator on stack.
+                 */
+                if(getPrecedenceOfOperators(stack1.top()) >= getPrecedenceOfOperators(expression_[i]))
+                {
+                    infixToPostfixExp = infixToPostfixExp + stack1.top();
+                    stack1.pop();
+                    stack1.push(expression_[i]);
+                }
+                else
+                {
+                    stack1.push(expression_[i]);
+                }
+            }
+        }
+        else
+        {
+            //if literal is operand, put it on to final postfix expression
+            infixToPostfixExp = infixToPostfixExp + expression_[i];
+        }
+        i++;
+    }
+
+    //poping out all remainig operator literals & adding to final postfix expression
+    if(!stack1.empty())
+    {
+        while(!stack1.empty())
+        {
+            infixToPostfixExp = infixToPostfixExp + stack1.top();
+            stack1.pop();
+        }
+    }
+
+    return infixToPostfixExp;
+
+}
+
+int main()
+{
+    InfixToPostfix p("a+b*c/d-q");
```

Comment:
Allow external input by user by declaring a string and then passing it to the object function.

Reply:
I had made the requested changes, please review implementation...